### PR TITLE
MBS-10822: Change tableColumns tables to use named parameters 

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-10822

Hopefully this makes things more readable and understandable, to begin with.

Also: a couple tiny fixes to related files.

### CODE REVIEW

File: root/utility/tableColumns.js [hunk 1]
```diff
@@ -282,14 +293,15 @@ export function defineInstrumentUsageColumn(
 }
 
 export function defineNameColumn<T: CoreEntityT | CollectionT>(
-  title: string,
-  order?: string = '',
-  sortable?: boolean = false,
-  descriptive?: boolean = true,
+  props: {
+    ...OrderableProps,
+    +descriptive: boolean,
```
Comment (yvanzo) [inline root/utility/tableColumns.js:298]:
```suggestion
    +descriptive?: boolean,
```

File: root/utility/tableColumns.js [hunk 2]
```diff
@@ -282,14 +293,15 @@ export function defineInstrumentUsageColumn(
...
+    +title: string,
+  },
 ): ColumnOptions<T, string> {
   return {
     Cell: ({row: {original}}) => (
-      descriptive
+      props.descriptive
```
Comment (yvanzo) [inline root/utility/tableColumns.js:304]:
```suggestion
    const descriptive = props.hasOwnProperty('descriptive') ? descriptive : true;
    return {
      Cell: ({row: {original}}) => (
        descriptive
```

File: root/components/list/AreaList.js [hunk 3]
```diff
@@ -38,13 +38,21 @@ const AreaList = ({
...
     () => {
       const checkboxColumn = $c.user && (checkboxes || mergeForm)
-        ? defineCheckboxColumn(checkboxes, mergeForm)
+        ? defineCheckboxColumn({mergeForm: mergeForm, name: checkboxes})
         : null;
-      const nameColumn =
-        defineNameColumn<AreaT>(l('Area'), order, sortable);
-      const typeColumn = defineTypeColumn('area_type', order, sortable);
+      const nameColumn = defineNameColumn<AreaT>({
+        descriptive: true,
```
Comment (yvanzo) [inline root/components/list/AreaList.js:44]:
```suggestion
```
and all other occurrences.

File: root/utility/tableColumns.js (hunk 2 above)
Comment (reosarevok) [reply root/utility/tableColumns.js:304]: Is this better though?

I feel defaulting things to true is confusing, so I thought specifying when we do want descriptive linking seemed more clear.

On the other hand, it's true that we want descriptive linking in most cases, so maybe the clearest option of them all would be change the condition from "descriptive" to "simpleLink" or something like that, and then only pass it when we specifically want to deviate from the (descriptive) norm?

Comment (yvanzo) [reply root/utility/tableColumns.js:304]: Default value can be either `true` or `false`, in either case that will be far better than requiring to set it for the vast majority of calls. “simpleLink” is simply not as descriptive as its negation “descriptive”, but there might be a clearer wording.

Comment (reosarevok) [reply root/utility/tableColumns.js]: Ok, since it seems it's just me who finds this kinda weird, then let's do it :)

**Comment by yvanzo** [comment]
@brainzbot retest this please

**Pull request merged**